test/unit/BCoWPool/BCoWPool.t.sol

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import {IERC20} from '@cowprotocol/interfaces/IERC20.sol';
+
+import {BCoWPoolBase} from './BCoWPoolBase.sol';
+
+import {IBCoWFactory} from 'interfaces/IBCoWFactory.sol';
+import {IBPool} from 'interfaces/IBPool.sol';
+
+contract BCoWPoolFinalize is BCoWPoolBase {
+  function setUp() public virtual override {
+    super.setUp();
+    bCoWPool.set__tokens(tokens);
+    bCoWPool.set__records(tokens[0], IBPool.Record({bound: true, index: 0, denorm: tokenWeight}));
+    bCoWPool.set__records(tokens[1], IBPool.Record({bound: true, index: 1, denorm: tokenWeight}));
+    bCoWPool.mock_call__mintPoolShare(INIT_POOL_SUPPLY);
+    bCoWPool.mock_call__pushPoolShare(address(this), INIT_POOL_SUPPLY);
+
+    vm.mockCall(address(this), abi.encodeCall(IBCoWFactory.logBCoWPool, ()), abi.encode());
+
+    vm.mockCall(tokens[0], abi.encodeCall(IERC20.approve, (vaultRelayer, type(uint256).max)), abi.encode(true));
+    vm.mockCall(tokens[1], abi.encodeCall(IERC20.approve, (vaultRelayer, type(uint256).max)), abi.encode(true));
+  }
+
+  function test_WhenCalled() external {
```

Comment:
test_What?

Reply:
is the result of implementing feedback from this: https://github.com/defi-wonderland/balancer-v1-amm/pull/159#discussion_r1680730282 :sob: what did you have in mind instead?

Reply:
is this because Finalize is the only method in the tree for now?

Reply:
yes, precisely. That's why PRs depending on this all have renames of these methods :sob: